client/src/shared/components/Table/index.tsx

```diff
@@ -1,73 +1,61 @@
-/* eslint-disable @typescript-eslint/strict-boolean-expressions */
-/* eslint-disable multiline-ternary */
-import React, { useState } from 'react';
-import type { ReactNode } from 'react';
-import HeaderTitle from './HeaderTitle';
+import React from 'react';
+import PropTypes from 'prop-types';
 
-export enum TableSortEnum {
-  ASC,
-  DESC
+interface Column {
+  accessor: string;
+  label: string;
 }
-export interface TableHeader {
-  text: string;
-  onClick?: () => void;
+interface TableProps {
+  data: Array<Record<string, any>>;
+  columns: Column[];
+  onClick: () => void;
 }
 
-export interface TableProps {
-  header: TableHeader[];
-  checkbox?: boolean;
-  children: ReactNode;
-  sortable?: boolean;
-}
-
-const Table: React.FC<TableProps> = ({
-  header,
-  children,
-  checkbox = true,
-  sortable = true,
-}: TableProps) => {
-  const [sort, setSort] = useState({
-    index: -1,
-    sortBy: TableSortEnum.ASC
-  });
-
-  const handleSortChange = (uid: number): void => {
-    setSort((curr) => ({
-      index: uid,
-      sortBy:
-        curr.index === uid
-          ? curr.sortBy === TableSortEnum.ASC
-            ? TableSortEnum.DESC
-            : TableSortEnum.ASC
-          : TableSortEnum.ASC,
-    }));
-  };
-
+const Table: React.FC<TableProps> = ({ data, columns, onClick }: TableProps) => {
   return (
-    <div className="overflow-auto text-neutral-900 flex justify-center rounded-[5px]">
-      <table className="text-left w-full">
-        <thead className="bg-neutral-200">
+    <div className="grid">
+      <table className="rounded-t-lg overflow-hidden">
+        <thead>
           <tr>
-            <th className="p-4">
-              {checkbox && <input type="checkbox" className="h-5 w-5 hover:bg-sky-700" />}
-            </th>
-            {header.map((item, index) => (
-              <th className="px-4 py-3 w-auto whitespace-nowrap text-sm" key={index}>
-                <HeaderTitle
-                  item={item}
-                  index={index}
-                  sort={sort}
-                  sortable={sortable}
-                  handleSortChange={handleSortChange}
-                />
+            {columns.map((column) => (
+              <th
+                className="bg-lightGray2 h-[49px] text-[15px] font-semibold text-left pl-3 pr-20"
+                key={column.accessor}
+              >
+                {column.label}
               </th>
             ))}
           </tr>
         </thead>
-        <tbody>{children}</tbody>
+        <tbody>
+          {data.map((item) => (
+            <tr
+              className={`${item.id % 2 === 0 ? '' : 'bg-gray-100'} align-middle cursor-pointer `}
+              key={item.id}
+              onClick={onClick}
+            >
```

Comment:
You can make use of `shared/components/Table/ListItem` component for the table rows since it already contains a column for `Actions`. 

You can just modify the row to be clickable with a prop i.e:
```suggestion
            <tr
              className={`${isClickable && 'cursor-pointer'} `}
              key={item.id}
              onClick={isClickable? onClick : undefined}
            >
```